compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1453,7 +1461,23 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         // frequently opened issues show.
         let opaque_ty_span = self.mark_span_with_reason(DesugaringKind::OpaqueTy, span, None);
 
-        let opaque_ty_def_id = self.local_def_id(opaque_ty_node_id);
+        let opaque_ty_def_id = match origin {
+            hir::OpaqueTyOrigin::TyAlias(def_id) => {
+                if let Some(def_id) = self.opt_local_def_id(opaque_ty_node_id) {
+                    def_id
+                } else {
+                    self.create_def(def_id, opaque_ty_node_id, DefPathData::ImplTrait)
+                }
+            }
+            hir::OpaqueTyOrigin::FnReturn(fn_def_id) => {
+                if let Some(def_id) = self.opt_local_def_id(opaque_ty_node_id) {
```

Comment:
I what case would this ever be `Some`?

Reply:
Same as https://github.com/rust-lang/rust/pull/102483#discussion_r984017996